jsonrpc/provider/src/provider.ts

```diff
@@ -54,7 +55,14 @@ export class JsonRpcProvider extends IJsonRpcProvider {
     request: RequestArguments<Params>,
     context?: any,
   ): Promise<Result> {
-    return this.requestStrict(formatJsonRpcRequest(request.method, request.params || []), context);
+    return this.requestStrict(
+      formatJsonRpcRequest(
+        request.method,
+        request.params || [],
+        getBigIntRpcId().toString() as any,
+      ), // casting to any is required in order to use BigInt as rpcId
```

Comment:
Maybe I'm missing your intention here, but wouldn't it make sense to update `formatJsonRpcRequest/formatJsonRpcResponse` and associated types to use `string | number` or strictly `string` to avoid plugging `any` here?

E.g. `formatJsonRpcRequest` still defaults to the lower entropy number id if not provided here: https://github.com/WalletConnect/walletconnect-utils/pull/76/files#diff-9cb8393a82bee49161c32626df17ec3a398b867f1d019f7a35b08a704f16ea29R21